### PR TITLE
SOLR-15966 Get rid of WEB-INF/lib/jsr305-3.0.2.jar

### PR DESCRIPTION
**Pull request by janhoy**
https://issues.apache.org/jira/browse/SOLR-15966

WIP attempt of excluding jsr305 and javax.annotations jars from the release.

I did a full `gradlew check` on this branch, and it succeeded, so I think it does no harm simply excluding these?

### CODE REVIEW

File: solr/modules/gcs-repository/build.gradle [hunk 1]
```diff
@@ -51,6 +56,7 @@ dependencies {
     implementation('io.opencensus:opencensus-api') { transitive = false }
     implementation('io.opencensus:opencensus-contrib-http-util') { exclude group: 'com.google.guava', module: 'guava' }
 
+    testImplementation('com.google.code.findbugs:jsr305') { transitive = false }
```
Comment (madrob) [inline solr/modules/gcs-repository/build.gradle:59]: I tried removing this and everything still seemed to work.

File: versions.lock [hunk 2]
```diff
@@ -29,7 +29,7 @@ com.google.cloud:google-cloud-core:1.94.3 (3 constraints: da2307c0)
 com.google.cloud:google-cloud-core-http:1.94.3 (2 constraints: 651543d8)
 com.google.cloud:google-cloud-storage:1.113.14 (2 constraints: 10143aa3)
 com.google.code.findbugs:annotations:3.0.1 (1 constraints: 0605fb35)
-com.google.code.findbugs:jsr305:3.0.2 (9 constraints: 057f5545)
+com.google.code.findbugs:jsr305:3.0.2 (2 constraints: cd195721)
```
Comment (dsmiley) [inline versions.lock:32]: What are the remaining 2 references?

File: solr/modules/gcs-repository/build.gradle (hunk 1 above)
Comment (janhoy) [reply solr/modules/gcs-repository/build.gradle:59]: I moved this from "implementation" to "testImplementation". But if it is not needed, then fine.

File: versions.lock (hunk 2 above)
Comment (janhoy) [reply versions.lock]: I excluded it from ref-guide and s3-repository as well, and now we're down to only

```
./gradlew why --hash=b70f3485
> Task :why
com.google.code.findbugs:jsr305:3.0.1
        com.google.code.findbugs:annotations -> 3.0.1
```

I think this one is the testOnly from https://github.com/apache/solr/blob/main/gradle/hacks/solr.findbugs.gradle#L40

**Comment by janhoy** [comment]
I think we got rid of it now from the distro
```
gw dev
find solr/packaging/build/dev | grep -E "jsr305|javax.annotation-api"
# (nothing)
```
Should these be "more global" in some way? Now each module will need to exclude.

But if no more improvements are suggested I'll merge this soon.

File: versions.lock [hunk 3]
```diff
@@ -29,7 +29,7 @@ com.google.cloud:google-cloud-core:1.94.3 (3 constraints: da2307c0)
 com.google.cloud:google-cloud-core-http:1.94.3 (2 constraints: 651543d8)
 com.google.cloud:google-cloud-storage:1.113.14 (2 constraints: 10143aa3)
 com.google.code.findbugs:annotations:3.0.1 (1 constraints: 0605fb35)
-com.google.code.findbugs:jsr305:3.0.2 (9 constraints: 057f5545)
+com.google.code.findbugs:jsr305:3.0.1 (1 constraints: b70f3485)
```
Comment (dsmiley) [inline versions.lock:32]: one remains...

Comment (janhoy) [reply versions.lock:32]: I think this is the testOnly inclusion mentioned earlier - the binary distro now does not include these jars. I think the test-only include is to support the annotations in IDEs

**Pull request merged**